external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -2601,36 +2587,38 @@ fn parse_dot_or_index_chain(context: &mut Context) -> Result<Exp, Box<Diagnostic
                             }
                         }
                     }
-                    _ => {
-                        let n = parse_identifier(context)?;
-                        if is_start_of_call_after_function_name(context, &n) {
-                            let call_start = context.tokens.start_loc();
-                            let is_macro = if let Tok::Exclaim = context.tokens.peek() {
-                                let loc = current_token_loc(context.tokens);
-                                context.tokens.advance()?;
-                                Some(loc)
+                    _ => match parse_identifier(context) {
+                        Err(diag) => {
+                            context.add_diag(*diag);
+                            Exp_::DotUnresolved(loc, Box::new(lhs))
+                        }
+                        Ok(n) => {
+                            if is_start_of_call_after_function_name(context, &n) {
```

Comment:
Do we want to check this either way so that we support `s.foo.(x, y)` ? I can imagine trying to get IDE help by hvaing `s.foo.bar(x, y)` and removing `bar` to try to get autocomplete.

Reply:
I think this will work without any changes as `s.foo.(x, y)` will parse to the `DotUnresolved` case and auto-completion will work correctly.

Reply:
Can you add a test?

Reply:
Added this to symbolicator tests, but I am also planning to have a more comprehensive auto-completion tests once auto-completion features start landing (i.e., have unit tests that actually check what would auto-completion actually insert in these partially-parsed cases).